crates/astria-conductor/tests/blackbox/firm_only.rs

```diff
@@ -377,3 +405,111 @@ async fn fetch_from_later_celestia_height() {
          within 1000ms",
     );
 }
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+async fn exits_on_celestia_chain_id_mismatch() {
+    use astria_grpc_mock::{
+        matcher,
+        response as GrpcResponse,
+        Mock as GrpcMock,
+    };
+
+    // We have to create our own test conductor and perform mounts manually because `TestConductor`
```

Comment:
Instead of doing all that lifting here, could you have made use of https://doc.rust-lang.org/std/mem/struct.ManuallyDrop.html to inhibit the custom `Drop` from being called?

Reply:
In attempting to do so, I'm not sure this method will work either. Calling `await` on the conductor contained by `ManuallyDrop<TestConductor>` errs because await takes ownership of the conductor. `ManuallyDrop` does have a method `take()` which would allow for this, but it is an unsafe function and as far as I know these can't be used in tests

Reply:
OK, let's leave this for now.